frontend_app_admin_workspace_user/src/component/AdminUser.jsx

```diff
@@ -104,8 +107,25 @@ export class AdminUser extends React.Component {
     if (resultSuccess > 0) this.handleToggleAddUser()
   }
 
+  filterUserList = () => {
+    const { props, state } = this
+    return state.userFilter === ''
+      ? props.userList
+      : props.userList.filter(usr => {
+        const userProfile = PROFILE_LIST.find(type => type.slug === usr.profile) || { label: '' }
+
+        return usr.public_name.toUpperCase().includes(state.userFilter.toUpperCase()) ||
+          usr.email.toUpperCase().includes(state.userFilter.toUpperCase()) ||
+          usr.username.toUpperCase().includes(state.userFilter.toUpperCase()) ||
+          props.t(userProfile.label).toUpperCase().includes(state.userFilter.toUpperCase()) ||
+          (props.t('Active').toUpperCase().includes(state.userFilter.toUpperCase()) && usr.is_active) ||
+          (props.t('Inactive').toUpperCase().includes(state.userFilter.toUpperCase()) && !usr.is_active)
```

Comment:
This is not wrong, but I find it a bit difficult to read, so I will make a suggestion but you can ignore it and mark this conversation as solved if you prefer the original version (and if you like it you can use the same logic to other functions)

```suggestion
        const hasFilterKeysOnPublicName = usr.public_name.toUpperCase().includes(state.userFilter.toUpperCase())
        const hasFilterKeysOnEmail = usr.email.toUpperCase().includes(state.userFilter.toUpperCase())
        const hasFilterKeysOnUsername = usr.username.toUpperCase().includes(state.userFilter.toUpperCase())
        const hasFilterKeysOnProfile = props.t(userProfile.label).toUpperCase().includes(state.userFilter.toUpperCase())
        const hasFilterKeysOnActif = (props.t('Active').toUpperCase().includes(state.userFilter.toUpperCase()) && usr.is_active) ||
          (props.t('Inactive').toUpperCase().includes(state.userFilter.toUpperCase()) && !usr.is_active)

        return hasFilterKeysOnPublicName || hasFilterKeysOnEmail || hasFilterKeysOnUsername || hasFilterKeysOnProfile || hasFilterKeysOnActif          
```

Reply:
This is indeed better IMO, I'll apply it on every functions I wrote or edited